File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if(not root):
            return []
        ll=[[root]]
        res=[[root.val]]
        # ll.append([1,2,3,4])
        
        # while(ll)
        # print(ll.pop())
        while(ll):
            pl=ll.pop()
            if(len(pl)==0):
                res.pop()
                return res
            res.append([])
            ll.append([])
            for p in pl:
                if(p.left):
                    ll[-1].append(p.left)
                    res[-1].append(p.left.val)
                if(p.right):
                    ll[-1].append(p.right)
                    res[-1].append(p.right.val)

        # res.pop()
        return res

File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    TreeNode* lowestCommonAncestor(TreeNode* root, TreeNode* p, TreeNode* q) {
        // if(!root){return nullptr;}
        int ip= p->val;
        int iq= q->val;
        while(root){
            cout<<root->val<<endl;
            if(root->val > ip && root->val > iq){ root=root->left;}
            else if(root->val < ip && root->val < iq){root=root->right;}
            else{return root;}
        }
        return root;
    }
};
